src/kernels/MIOpenPadConstant.cpp

```diff
@@ -0,0 +1,190 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <cstddef>
+#include <cstdio>
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_runtime.h>
+#include <hip/hip_fp16.h>
+#endif
+
+#include "float_types.h"
+#include "tensor_view_5d.hpp"
+
+template <typename TI, typename TO>
+__device__ void padconstantfwd(const TI* __restrict__ x,
+                               TO* __restrict__ y,
+                               const tensor_view_5d_t x_tv,
+                               const tensor_view_5d_t y_tv,
+                               const padding_5d_t padding,
+                               const size_t output_size,
+                               TO value)
+{
+    TO padding_value   = CVT_ACCUM2FLOAT(value);
+    const uint64_t gid = threadIdx.x + blockIdx.x * blockDim.x;
+    if(gid >= output_size)
+        return;
+
+    uint64_t o[5];
+    getNCDHW(o, gid, y_tv.size);
+    bool flag = true;
+
+    for(int i = 0; i < 5; ++i)
+    {
+        o[i] = o[i] - padding.val[2 * i];
+        flag *= o[i] < x_tv.size[i];
```

Comment:
I think it's a little different from moDNN's implementation. After subtracting the value from o[i], check whether o[i]>=0. In order to implement this, I think the types need to be singed rather than insigned.

Reply:
The values is also unsigned in moDNN, and since it's unsigned, `o[i]` is always guaranteed to be >= 0, rendering that check useless. That's why I removed it

Reply:
The intention of the kernel creator is to pad when o becomes negative. However, because o is unsinged, it is not negative and becomes the max value. And the condition o[i] < input_tv.size[i] becomes false, so I think the same result occurred by chance.
moDNN's kernel is created by us, so there may be mistakes. I'm sorry about this.
But, if this was intended and worked, it doesn't matter, but if you accidentally create an implementation that produces the same result, I don't think it's a good idea.
Although our work may be a relatively simple hip porting, it also involves revisiting and refining the existing moDNN kernels.
Therefore, when adding this kernel to miopen, I think it would be more correct to add a comment that the unsinged variable becomes negative and the same implementation is achieved, or to implement singed using a variable.
I recommend the latter because I think it is better to keep comments to a minimum, but this may be a style thing.
I'm sorry that the PR review took so long, but I think it will be of great help to future cooperation.
Thank you.

Reply:
I've changed `o[i]` in the kernel to be `int64_t` and modified the supporting inline functions. Tests so far seems to suggest that the kernel is still behaving as expected. Please check and see if there are any issues I am not aware of